hack/check-go-version.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash -e
+#
+# Copyright 2025 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+allowed_go_version=1.22
```

Comment:
Is this for downstream ?

Reply:
Can we add this information to go.mod with a comment, so it's next to the golang deps source of truth ? 
like at go.mod with a comment
```
odule github.com/k8snetworkplumbingwg/kubemacpool                              

// allowed_go=1.22                                                                                
go 1.22     
```

then we parse the go.mod with something like
```bash
allowed_go=$(grep allowed_go go.mod |awk '{print $3}')
```

Reply:
+1!